### PR TITLE
DOT-4440: Fix a bug when we get data by transaction ID

### PR DESCRIPTION
**Pull request by olyans**
https://nextmediama.atlassian.net/browse/DOT-4440

### CODE REVIEW

File: src/API/HTTPAdapter/CurlHTTPAdapter.php [hunk 1]
```diff
@@ -35,16 +35,21 @@ protected function parseHeaders($headers)
...
-        $curl = curl_init(sprintf('%s%s', $this->getBaseUrl(), $endpoint));
-        $headers["Content-Type"] = "application/json";
         $headers["Accept"] = "application/json";
+        $method = strtolower($method);
+
+        if ($method == 'post') {
+            $curl = curl_init(sprintf('%s%s', $this->getBaseUrl(), $endpoint));
+        } else {
+            $curl = curl_init(sprintf( '%s%s?%s', $this->getBaseUrl(), $endpoint, http_build_query($params)));
+        }
```
Comment (kitro) [inline src/API/HTTPAdapter/CurlHTTPAdapter.php:45]: Missing space in 
`$curl = curl_init(sprintf( '%s%s?%s', $this->getBaseUrl(), $endpoint, http_build_query($params)));` 
should be
 `$curl = curl_init(sprintf('%s%s?%s', $this->getBaseUrl(), $endpoint, http_build_query($params)));`

File: src/API/HTTPAdapter/CurlHTTPAdapter.php [hunk 2]
```diff
@@ -35,16 +35,21 @@ protected function parseHeaders($headers)
 
     public function request(string $method, string $endpoint, array $params = [], array $headers = []): Response
     {
-        $curl = curl_init(sprintf('%s%s', $this->getBaseUrl(), $endpoint));
-        $headers["Content-Type"] = "application/json";
         $headers["Accept"] = "application/json";
+        $method = strtolower($method);
+
+        if ($method == 'post') {
```
[the remark this answers is not in the document]
Comment (olyans) [reply src/API/HTTPAdapter/CurlHTTPAdapter.php]: https://github.com/NextmediaMa/youcan-payment-php-sdk/pull/8/commits/f201c09c833bed16dc80f758d6f400dad00b7aad

File: src/API/HTTPAdapter/CurlHTTPAdapter.php (hunk 1 above)
Comment (olyans) [reply src/API/HTTPAdapter/CurlHTTPAdapter.php:45]: https://github.com/NextmediaMa/youcan-payment-php-sdk/pull/8/commits/cd29a3fc2cb76a2fd7606af77fd683ff78e8d7aa

**Review by kitro — APPROVED** [review]
Looks good to me :+1: :+1:

**Pull request merged**